simpleviewinc/keg-hub
# keg-components: Text toggle component

**Ticket**: [issue_id](https://jira.simpleviewtools.com/browse/ZEN-391)
`[semver]: minor`

## Context

* The Session details on mobile has this expandable text component
    * ![image](https://user-images.githubusercontent.com/3317835/95791650-aa7eda80-0c96-11eb-9341-f8940c20a8c2.png)

* we currently don't have a component like that
* create a component that can do this:
   * cuts off text given a max # of lines
   * has a toggle (show more/less) button
   * when the toggle button is clicked, show all the text 


## Goal

* create `TextToggle` component that can do those things stated in **Context**
* add the appropriate stories and examples
* utilize `Drawer` component within the `Text Toggle` component
* update `Drawer` component slightly so we can set the collapsed height at any point, as opposed to first render

## Updates

**Keg-Components**
* installed `react-native-linear-gradient` and `react-native-web-linear-gradient`

* `repos/keg-components/src/components/textToggle/textToggle.js`
   * TextToggle component
* `repos/keg-components/src/components/textToggle/textToggle.stories.mdx`
   * stories for textToggle
* `repos/keg-components/src/components/textToggle/textToggle.examples.js`
    * examples for TextToggle that's utilized in the stories.mdx

* `repos/keg-components/configs/babel.config.js`
   * added the dependencies to ensure the props table work - same as the one @mpcarolin used in his PR

* `repos/keg-components/src/components/drawer/drawer.examples.js`
   * no changes, husky formatting

* `repos/keg-components/src/components/drawer/drawer.js`
   * eslint formatting
   * updated `initial` prop to be `collapsedHeight`
   * also updated so we can set the collapsedHeight at any point as opposed to only on first render

**Keg-Core**
* added dev dependency for `react-native-linear-gradient`


## Testing

* Run `keg dpg run package=docker.pkg.github.com/simpleviewinc/keg-packages/keg-components:text-toggle-component`
* navigate to http://local.kegdev.xyz/?path=/docs/interactions-texttoggle--basic
* verify the toggling of the text works
